src/store/channels/saga.ts

```diff
@@ -62,9 +63,20 @@ export function* clearChannels() {
   yield put(removeAll({ schema: schema.key }));
 }
 
+function* listenForChannelLoadedEvent() {
+  const channel = yield call(conversationsChannel);
+  while (true) {
+    const { channelId = undefined } = yield take(channel, '*');
```

Comment:
I'm trying to decide if this is the right event. If we start pre-loading messages for some reason, for example, then we'd be marking the messages as read all incorrectly for some things. Maybe we just need to make this more strict and only react if the channel matches the open conversation / channel?

Reply:
Also just had another thought, we also need to mark as read when new real time messages come in and the conversation is open.

Reply:
I know about the `activeConversationId`, but i am not sure if we have an "active channel id"..? Or can we just extract/know that from the url maybe? eg

![image](https://github.com/zer0-os/zOS/assets/33264364/4b4b8a3f-00dd-46cd-89eb-6879b0c193a8)

Reply:
Yeah, I'm not sure, we'd have to dig around.  But somehow we must know because we load the messages for the channel and receive real time messages, etc. Maybe it mostly currently works just based on the fact that it's open? We should probably have that in state somewhere.

Reply:
So this is getting the channelId from the current url i think, and passing it via props in the UI

https://github.com/zer0-os/zOS/blob/a51af1a8074d6efe3cdc68ad5d657066e5282a55/src/platform-apps/channels/index.tsx#L19

> But somehow we must know because we load the messages for the channel and receive real time messages, etc

correct.

Reply:
I will update to read the "current channel id" from the url. And also trigger `mark-as-read` on receiveNewMessage sendbird event

Reply:
In this case I think maybe we need to have the component tell us when the channel changes. All the routing is done via React so it makes sense that it is the original source of that state. I.E., when the channel changes in the main component (wherever that is) it can call a "setCurrentChannel" or something that will just update the global state.

Reply:
I have updated the PR as per your comments. But the code got a little more complex that i would like 🤔